kem/src/lib.rs

```diff
@@ -9,11 +9,24 @@
 #![forbid(unsafe_code)]
 #![warn(missing_docs, unused_qualifications, missing_debug_implementations)]
 
-#[cfg(feature = "std")]
-extern crate std;
+/// A value that can be encapsulated to. Often, this will just be a public key. However, it can
+/// also be a bundle of public keys, or it can include a sender's private key for authenticated
+/// encapsulation.
+pub trait Encapsulate<EK, SS> {
+    /// Encapsulation error
+    type Error;
 
-mod errors;
-mod kem;
+    /// Encapsulates a fresh shared secret
+    fn encapsulate(&self, rng: impl rand_core::CryptoRngCore) -> Result<(EK, SS), Self::Error>;
```

Comment:
It seems to me that `&mut impl rand_core::CryptoRngCore` is the right answer here.  All the methods that let you get any use out of `CryptoRngCore` take `&mut self`, so at the very least, you need a mutable thing here.  And there's no need to pass ownership of the RNG to this function.

Reply:
Yeah, `&mut impl CryptoRngCore` is the convention we've selected to use in #1148